### ISSUE TITLE
mod_archive_odbc

### ISSUE THREAD

**Issue by ikawka** [processone/ejabberd-contrib#15]
Hello,

I've managed to have this compiled and loaded the module.

here's from the configuration file
```
mod_archive_odbc:
    database_type: mysql
    default_auto_save: true
    enforce_default_auto_save: true
    default_expire: "infinity"
    enforce_min_expire: 0
    enforce_max_expire: "infinity"
    replication_expire: 31536000
```
but after restarting the ejabberd i got this error
```
2014-03-07 16:01:08.866 [error] <0.9739.0> CRASH REPORT Process <0.9739.0> with 0 neighbours exited with reason: no case clause matching undefined in gen_iq_handler:add_iq_handler/6 line 65 in gen_server:init_it/6 line 328
```

hoping you guys could shed some light on the one. thank you very much

**Comment by mprokop87** [comment]
Has there been any fixes for this issue? I'm using 14.12 and experience it as well.

**Comment by weiss** [comment]
`mod_archive` has not been updated to work with recent ejabberd releases, see the [note at the top of the README file][1].  I'm sure patches would be welcome, but I for one am not too motivated to put work into `mod_archive`, as the [archiving extension][2] it implements is somewhat deprecated and slowly being replaced by [MAM][3] in practice (and there's an [ejabberd module][4] for that).

[1]: https://github.com/processone/ejabberd-contrib/blob/53ca8bafba45dda146d5f909edfa6425a54f9ada/mod_archive/README.txt#L3
[2]: http://xmpp.org/extensions/xep-0136.html
[3]: http://xmpp.org/extensions/xep-0313.html
[4]: https://github.com/processone/ejabberd-contrib/tree/master/mod_mam

**Comment by mprokop87** [comment]
I saw mam but did not see a way to configure it with postgres.  Is there
somewhere you can point me? The project I an working in requires it to
store messages in the pgsql db.

Thanks Matt

**Comment by weiss** [comment]
Ah right, the MAM module doesn't support SQL backends yet.  Unlike with `mod_archive`, I'd be interested in adding support, though.  This is nothing that will be ready tomorrow, but if you have a little time and would be interesting in testing new code, I'd have a go at it.

**Comment by mprokop87** [comment]
I would definitely be interested in helping out. I have around a month to month and a half before I need this portion of the project finished. I will shoot you an email from my work email. Thanks!

**Comment by mremond** [comment]
I will check how we can merge and contribute our own module. I will keep you posted fast.

**Comment by thomasbiddle** [comment]
So to clarify - I'm in using v15.11; I have no way of enabling mod_archive or mod_archive_* - And as such, am not able to store history on my server?

Am I understanding that correctly?

**Comment by weiss** [comment]
> I will check how we can merge and contribute our own module.

As this happened with the [15.06 release][1], and as the [extension implemented by `mod_archive`][2] doesn't seem to be popular these days, I guess this issue could be closed.

[1]: https://blog.process-one.net/ejabberd-15-06/
[2]: http://xmpp.org/extensions/xep-0136.html

**Comment by weiss** [comment]
> If I'm using v15.11; I have no way of enabling `mod_archive` or `mod_archive_*`

Right.

>  And as such, am not able to store history on my server?

You could use `mod_mam` (as shipped with ejabberd 15.11) instead.  This implements [XEP-0313][1], which is what people use these days.

[1]: http://xmpp.org/extensions/xep-0313.html
